osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -243,7 +244,7 @@ public async Task<TModel> Import(string path, CancellationToken cancellationToke
         /// </summary>
         /// <param name="archive">The archive to be imported.</param>
         /// <param name="cancellationToken">An optional cancellation token.</param>
-        public Task<TModel> Import(ArchiveReader archive, CancellationToken cancellationToken = default)
+        public async Task<TModel> Import(ArchiveReader archive, CancellationToken cancellationToken = default)
```

Comment:
why?

Reply:
There is a path which returns null, which will crash when awaited.

Reply:
sounds like the issue is elsewhere (the null path).

and i don't see how adding `async-await` pair here changes anything. it should not be done for the above explanation.

Reply:
The handling for a null result of CreateModel was already there, it would just have thrown an exception when the method was awaited (since it doesn't return a Task that yields null when awaited, but an actual null). This makes it so that when a null is returned, it will work as intended.

Reply:
I could also have changed the `return null` to `return new Task(() => null);`, but this seemed like a cleaner solution

Reply:
`Task.FromResult(null)` is an alternative